### PR TITLE
added signicat/kvk scheme

### PR DESCRIPTION
**Pull request by michielmayer**

### CODE REVIEW

**Comment by bobhageman** [comment]
Hi @michielmayer, after reviewing I've a few questions:
- In our opinion the CredentialId `official` and thus folder naming is not very descriptive. Think of a third party developer who wants to integrate Signicat kvk issued credentials in their software: they would have to refer to attributes like `signicat.official.tradeNames` for example. This does not make clear it's a KVK issued attribute. And what if Signicat would like to issue credentials from other issuers in the near future? Then you would get other credential id's / folders next to the `official` folder. How would you distinguish them?
- Some of the attributes, like `rsin` are marked optional. To be sure: this is done on purpose?

**Comment by michielmayer** [comment]
- I changed it to 'kvkTradeRegister'. I think that makes more sense indeed
- I marked attributes as optional, when it is possible that they are empty. In the issuance process I set all values that are not applicable to empty strings. I realize that some of the attributes that we marked as optional, the verifier really wants to know that they are empty. (eg. the verifier wants to know that specialLegalSituation is empty) What do you recommend in this case?

**Comment by michielmayer** [comment]
in addition to my last bullet rsin for example, is not applicable for 'eenmanszaken', only for other legal entities, so that attribute is correctly set to optional. Am I right?

**Comment by bobhageman** [comment]
Nice one, I think `kvkTradeRegister` is a better fit. Please also change the credential folder name `official` into `kvkTradeRegister`. These two need to be exactly the same.

About the optional attributes:
Note that what we're doing here is primary relevant to the issuing side. So if an attribute is marked as optional, it can be left out by KVK during issuance. Otherwise, if a missing attribute is not marked as optional, the app won't accept it when retrieving/storing the credential in the app. Summarized:
- If an attribute is marked optional: it can be left out (missing in total) by the issuing party.
- If an attribute is not marked optional: it cannot be left out and it must have a value, in emptiest form just a blank string. Never the value "NULL".

About your questions:
1. If the verifier wants to know specialLegalSituation is empty or not, it cannot be marked as optional. This way it must always be provided by KVK with either a value "abc" or as an empty string "". The verifier then knows it is always provided whether it's empty or not.
2. Regarding the rsin attribute I think it is correct to set it as optional, since it is not always provided.

**Comment by michielmayer** [comment]
Ok, thanks. I updated the optional fields according to your recommendations

**Comment by bobhageman** [comment]
Hi @michielmayer, discovered your key isn't capable of handling 44 attributes. Please generate a new keypair and replace the public key.

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 1]
```diff
@@ -0,0 +1,463 @@
+<IssueSpecification version="1">
```
Comment (bobhageman) [inline signicat/Issues/kvkTradeRegister/description.xml:1]: Please change to `version="4"`

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 2]
```diff
@@ -0,0 +1,463 @@
...
+			</Description>
+		</Attribute>
+		<Attribute id="legalEntity">
+			<Name>
+				<en>Legal entity</en>
+				<nl>Rechtsvorm</nl>
+			</Name>
+			<Description>
+				<en>Legal entity of the owner of the organisation</en>
+				<nl>Rechtsvorm van de eigenaar van de maatschappelijke activiteit </nl>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:79]: remove blank space at the end

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+			</Description>
+		</Attribute>
+		<Attribute id="address">
+			<Name>
+				<en>Address</en>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:84]: "Address of organisation" to be in line with the rest below?

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 4]
```diff
@@ -0,0 +1,463 @@
...
+				<nl>Emailadres van organisatie</nl>
+			</Name>
+			<Description>
+				<en>Email address of the organisation</en>
+				<nl>Emailadres van een maatschappelijke activiteit</nl>
+			</Description>
+		</Attribute>
+		<Attribute id="phone" optional="true">
+			<Name>
+				<en>Phone of organisation</en>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:104]: "the organisation" to be in line with the rest, same for dutch translation

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 5]
```diff
@@ -0,0 +1,463 @@
...
+				<nl>De bijzondere rechtstoestand van de eigenaar van de maatschappelijke activiteit, zoals: Faillissement, schuldsanering of surseance van betaling</nl>
+			</Description>
+		</Attribute>
+		<Attribute id="restrictionInLegalAction">
+			<Name>
+				<en>Restriction in legal action</en>
+				<nl>Beperking in rechtshandeling</nl>
+			</Name>
+			<Description>
+				<en>The restriction in legal action of the owner of the organisation, eg: Provisional rule, placed under guardianship of placed under administration</en>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:168]: 'of placed' -> 'or placed'

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 6]
```diff
@@ -0,0 +1,463 @@
...
+				<nl>De bijzondere rechtstoestand van de functionaris, zoals: Faillissement, schuldsanering of surseance van betaling</nl>
+			</Description>
+		</Attribute>
+		<Attribute id="restrictionInLegalActionOfficial">
+			<Name>
+				<en>Restriction in legal action of the official</en>
+				<nl>Beperking in rechtshandeling functionaris</nl>
+			</Name>
+			<Description>
+				<en>The restriction in legal action of the official, zoals: Provisional rule, placed under guardianship of placed under administration</en>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:388]: "zoals" -> "eg"

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 7]
```diff
@@ -0,0 +1,463 @@
...
+		<nl>The persoonlijke informatie van een functionaris voor een rechtspersoon</nl>
+	</Description>
+	<IssueURL>
+		<en>https://organisatiegegevens.signicat.nl</en>
+		<nl>https://organisatiegegevens.signicat.nl</nl>
+	</IssueURL>
+	<Attributes>
+		<Attribute id="kvkNumber">
+			<Name>
+				<en>KVK Number</en>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:24]: Number should be written without capital, same below for dutch translation

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 8]
```diff
@@ -0,0 +1,463 @@
...
+			</Name>
+			<Description>
+				<en>Email address of the organisation</en>
+				<nl>Emailadres van een maatschappelijke activiteit</nl>
+			</Description>
+		</Attribute>
+		<Attribute id="phone" optional="true">
+			<Name>
+				<en>Phone of organisation</en>
+				<nl>Telefoon van organisatie</nl>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:105]: Gekke Nederlandse naam,. misschien gewoon "Telefoonnummer"? Dezelfde hierboven voor en translation.

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 9]
```diff
@@ -0,0 +1,463 @@
...
+			</Name>
+			<Description>
+				<en>Full name of the official</en>
+				<nl>Volledige naam van de functionaris</nl>
+			</Description>
+		</Attribute>
+		<Attribute id="typeOfficial">
+			<Name>
+				<en>Type of official</en>
+				<nl>Type Functionaris</nl>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:245]: Functionaris zonder hoofdletter

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 10]
```diff
@@ -0,0 +1,463 @@
...
+			</Name>
+			<Description>
+				<en>Type of official</en>
+				<nl>Type Functionaris</nl>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:249]: Functionaris zonder hoofdletter

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 11]
```diff
@@ -0,0 +1,463 @@
...
+				<nl>Handlichting</nl>
+			</Name>
+			<Description>
+				<en>Emancipation of a minor</en>
+				<nl>Handlichting houdt in dat een minderjarige door toewijzing van een rechter dezelfde bevoegdheden toegekend krijgt als een meerderjarige.</nl>
+			</Description>
+		</Attribute>
+		<Attribute id="hasRestriction">
+			<Name>
+				<en>Has restrictions</en>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:424]: Engels is in meervoud en nederlands niet?

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 12]
```diff
@@ -0,0 +1,463 @@
...
+				<nl>Heeft beperking</nl>
+			</Name>
+			<Description>
+				<en>Indication of whether there is some kind of restriction that affects the determination of an authorization</en>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:428]: Engels mist "yes" "no" in lijn met de rest hierboven?

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 13]
```diff
@@ -0,0 +1,463 @@
...
+				<nl>Indicatie of er sprake is van enige vorm van beperking, die van invloed is op het bepalen van een bevoegdheid ('Ja', 'Nee')</nl>
```
Comment (stenreijers) [inline signicat/Issues/kvkTradeRegister/description.xml:429]: de apostrof bij "ja" en "nee" kan weg zoals hierboven ook niet gedaan is? In iedergeval suggestie om recht te trekken :).

**Review by stenreijers — COMMENTED** [review]
I have checked the translations on names and descriptions. Here are a few suggestions that you can do to improve upon the initial version :)

**Comment by michielmayer** [comment]
Thanks for comments, Sten. I made the changes in the last commit. @bobhageman: I also updated the version numbers

**Comment by michielmayer** [comment]
I am waiting for Gert-Jan for a new public key, that can handle 50 attributes. I will push it, as soon as I have got it

**Comment by michielmayer** [comment]
new public key is pushed

**Comment by ivard** [comment]
> Thanks for comments, Sten. I made the changes in the last commit. @bobhageman: I also updated the version numbers

I think you didn't resolve all comments of @stenreijers yet. Are you planning to still do that or what's the current status?

**Comment by michielmayer** [comment]
Thanks, I missed a few text changes indeed. They should all be in now.

File: signicat/PublicKeys/0.xml [hunk 14]
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<IssuerPublicKey xmlns="http://www.zurich.ibm.com/security/idemix">
+   <Counter>0</Counter>
+   <ExpiryDate>1738581417</ExpiryDate>
```
Comment (bobhageman) [inline signicat/PublicKeys/0.xml:4]: @michielmayer, missed this during my previous review, but I noticed the timestamp at `ExpiryDate` is set to 2 years from now. This should be 1 year.

You can easily change this value by the way, no need to generate a new keypair

Comment (michielmayer) [reply signicat/PublicKeys/0.xml]: ok, I changed the expiry date to 1 year from now

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 15]
```diff
@@ -0,0 +1,463 @@
...
+			</Description>
+		</Attribute>
+		<Attribute id="typeOwner">
+			<Name>
+				<en>Type of owner</en>
+				<nl>Type eigenaar</nl>
+			</Name>
+			<Description>
+				<en>The type of owner of the organisation</en>
+				<nl>De type eigenaar van de maatschappelijke activiteit</nl>
```
Comment (sanderhollaar) [inline signicat/Issues/kvkTradeRegister/description.xml:69]: 'Het' type

Some 'description' fields start with an article, some not. To make this more consistent maybe remove the first article' in all description-fields

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 16]
```diff
@@ -0,0 +1,463 @@
...
+			</Description>
+		</Attribute>
+		<Attribute id="specialLegalSituation">
+			<Name>
+				<en>Special legal situation</en>
+				<nl>Bijzondere rechtstoestand</nl>
+			</Name>
+			<Description>
+				<en>The special legal situation of the owner of the organisation, eg: Bankruptcy, debt restructuring or suspension of payment</en>
+				<nl>De bijzondere rechtstoestand van de eigenaar van de maatschappelijke activiteit, zoals: Faillissement, schuldsanering of surseance van betaling</nl>
```
Comment (sanderhollaar) [inline signicat/Issues/kvkTradeRegister/description.xml:159]: incorrect use of capital in 'Bankruptcy' and 'Faillissement'

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 17]
```diff
@@ -0,0 +1,463 @@
...
+			</Description>
+		</Attribute>
+		<Attribute id="restrictionInLegalAction">
+			<Name>
+				<en>Restriction in legal action</en>
+				<nl>Beperking in rechtshandeling</nl>
+			</Name>
+			<Description>
+				<en>The restriction in legal action of the owner of the organisation, eg: Provisional rule, placed under guardianship or placed under administration</en>
+				<nl>De beperking in rechtshandeling van de eigenaar van de maatschappelijke activiteit, zoals: Provisioneel bewind, onder curatele gesteld of onder bewind gesteld</nl>
```
Comment (sanderhollaar) [inline signicat/Issues/kvkTradeRegister/description.xml:169]: incorrect use of capital in 'Provisional' and 'Provisioneel'

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 18]
```diff
@@ -0,0 +1,463 @@
...
+			</Name>
+			<Description>
+				<en>Last name of the official</en>
+				<nl>Geslachtsnaam van de functionaris</nl>
+			</Description>
+		</Attribute>
+		<Attribute id="prefix">
+			<Name>
+				<en>Prefix</en>
+				<nl>Voorvoegsel geslachtsnaam</nl>
```
Comment (sanderhollaar) [inline signicat/Issues/kvkTradeRegister/description.xml:195]: ~geslachtsnaam~

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 19]
```diff
@@ -0,0 +1,463 @@
...
+			</Description>
+		</Attribute>
+		<Attribute id="specialLegalSituationOfficial">
+			<Name>
+				<en>Special legal situation of the official</en>
+				<nl>Bijzondere rechtstoestand functionaris</nl>
+			</Name>
+			<Description>
+				<en>The special legal situation of the official, eg: Bankruptcy, debt restructuring or suspension of payment</en>
+				<nl>De bijzondere rechtstoestand van de functionaris, zoals: Faillissement, schuldsanering of surseance van betaling</nl>
```
Comment (sanderhollaar) [inline signicat/Issues/kvkTradeRegister/description.xml:379]: incorrect use of capital in 'Bankruptcy' and 'Faillissement'

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 20]
```diff
@@ -0,0 +1,463 @@
...
+			</Description>
+		</Attribute>
+		<Attribute id="restrictionInLegalActionOfficial">
+			<Name>
+				<en>Restriction in legal action of the official</en>
+				<nl>Beperking in rechtshandeling functionaris</nl>
+			</Name>
+			<Description>
+				<en>The restriction in legal action of the official, eg: Provisional rule, placed under guardianship or placed under administration</en>
+				<nl>De beperking in rechtshandeling van de functionaris, zoals: Provisioneel bewind, onder curatele gesteld of onder bewind gesteld</nl>
```
Comment (sanderhollaar) [inline signicat/Issues/kvkTradeRegister/description.xml:389]: incorrect use of capital in 'Provisional' and 'Provisioneel'

File: signicat/Issues/kvkTradeRegister/description.xml [hunk 21]
```diff
@@ -0,0 +1,463 @@
...
+			</Name>
+			<Description>
+				<en>The reason for (not) being authorized or not being able to determine someone's authorization</en>
+				<nl>De reden voor het niet/wel bevoegd zijn of niet kunnen vastellen van iemands bevoegdheid</nl>
+			</Description>
+		</Attribute>
+		<Attribute id="referenceMoment">
+			<Name>
+				<en>Reference moment</en>
+				<nl>Peilmoment (opvraagmoment uit handelsregister)</nl>
```
Comment (sanderhollaar) [inline signicat/Issues/kvkTradeRegister/description.xml:455]: ~(opvraagmoment uit handelsregister)~

**Review by sanderhollaar — COMMENTED** [review]
minor feedback on copy

File: signicat/Issues/kvkTradeRegister/description.xml (hunk 21 above)
Comment (michielmayer) [reply signicat/Issues/kvkTradeRegister/description.xml:455]: deze heb ik laten staan, want dat willen we vanuit kvk extra duidelijk hebben

**Comment by michielmayer** [comment]
Thanks @sanderhollaar, I made the text changes, except for one. see inline comment.

**Review by sanderhollaar — APPROVED** [review]

**Pull request closed without merging**